### PR TITLE
downgrade.feature: update dnf4 scenarios for dnf5

### PR DESCRIPTION
**Pull request by glum23**
Updated dnf4 scenarios for dnf5 in downgrade.feature

### CODE REVIEW

File: dnf-behave-tests/dnf/downgrade.feature [hunk 1]
```diff
@@ -87,16 +88,15 @@ Scenario: Downgrade a package that was installed via rpm
         | Replaced  | flac-0:1.3.3-3.fc29.x86_64 | External User | @System               |
 
 
-# @dnf5
-# TODO(nsella) different exit code 0
-# TODO(nsella) different stderr
+@dnf5
+@not.with_dnf=4
```
[the remark this answers is not in the document]
Comment (glum23) [reply dnf-behave-tests/dnf/downgrade.feature:92]: I suppose those tags will be removed globally in this branch when dnf4 won't be the primary SW manager on Fedora. However, now it is useful to distinguish scenarios that work both for dnf5+dnf4 from those that work only for one of them.

File: dnf-behave-tests/dnf/downgrade.feature [hunk 2]
```diff
@@ -87,16 +88,15 @@ Scenario: Downgrade a package that was installed via rpm
...
 Scenario Outline: Check <command> exit code - package does not exist
   Given I use repository "dnf-ci-fedora"
    When I execute dnf with args "<command> non-existent-package"
-   Then the exit code is 1
+   Then the exit code is 0
```
[the remark this answers is not in the document]
Comment (glum23) [reply dnf-behave-tests/dnf/downgrade.feature:96]: I've asked about it in https://github.com/rpm-software-management/dnf5/issues/137, no answer yet.

[the remark this answers is not in the document]
Comment (j-mracek) [reply dnf-behave-tests/dnf/downgrade.feature:96]: I also think that this is a bug in DNF5. May I ask you to not modify the test and track it as a bug?

File: dnf-behave-tests/dnf/downgrade.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (glum23) [reply dnf-behave-tests/dnf/downgrade.feature]: OK, I've discussed it with Jarda, removing tags.

File: dnf-behave-tests/dnf/downgrade.feature (hunk 2 above)
[the remark this answers is not in the document]
Comment (glum23) [reply dnf-behave-tests/dnf/downgrade.feature]: OK, created issue: https://github.com/rpm-software-management/dnf5/issues/524
The exit code has been updated in the test.

[the remark this answers is not in the document]
Comment (kontura) [reply dnf-behave-tests/dnf/downgrade.feature]: The issue in dnf5 should be resolved by: https://github.com/rpm-software-management/dnf5/pull/532
Can you fix the test so it expects 1.

[the remark this answers is not in the document]
Comment (glum23) [reply dnf-behave-tests/dnf/downgrade.feature]: hmm, I've updated the test, let's see whether all occurrences were appropriately fixed

File: dnf-behave-tests/dnf/downgrade.feature [hunk 3]
```diff
@@ -24,10 +23,10 @@ Scenario: Downgrade one RPM
         | Action        | Package                                   |
         | downgrade     | flac-0:1.3.3-1.fc29.x86_64                |
    When I execute dnf with args "downgrade flac"
-   Then the exit code is 0
-    And stderr contains "Package flac of lowest version already installed, cannot downgrade it."
+   Then the exit code is 1
```
Comment (kontura) [inline dnf-behave-tests/dnf/downgrade.feature:26]: dnf5 has the same behavior as dnf4 so during a downgrade if the package is already at the lowest version the exit code is 0.

File: dnf-behave-tests/dnf/downgrade.feature [hunk 4]
```diff
@@ -87,96 +85,44 @@ Scenario: Downgrade a package that was installed via rpm
...
    When I execute dnf with args "upgrade flac"
-   Then the exit code is 0
-    And stdout is
-    """
-    <REPOSYNC>
-    Dependencies resolved.
-    Nothing to do.
-    Complete!
-    """
+   Then the exit code is 1
```
Comment (kontura) [inline dnf-behave-tests/dnf/downgrade.feature:118]: dnf5 has the same behavior as dnf4 so during an upgrade if the package is already at the highest version the exit code is 0.

File: dnf-behave-tests/dnf/downgrade.feature [hunk 5]
```diff
@@ -87,96 +85,44 @@ Scenario: Downgrade a package that was installed via rpm
...
-    <REPOSYNC>
-    Dependencies resolved.
-    Nothing to do.
-    Complete!
-    """
-    And stderr is
-    """
-    Package flac of lowest version already installed, cannot downgrade it.
-    """
+   Then the exit code is 1
```
Comment (kontura) [inline dnf-behave-tests/dnf/downgrade.feature:127]: dnf5 has the same behavior as dnf4 so during a downgrade if the package is already at the lowest version the exit code is 0.

**Review by kontura — CHANGES_REQUESTED** [review]

File: dnf-behave-tests/dnf/downgrade.feature (hunk 5 above)
Comment (glum23) [reply dnf-behave-tests/dnf/downgrade.feature:127]: depends on https://github.com/rpm-software-management/dnf5/issues/582

**Comment by glum23** [comment]
depends on https://github.com/rpm-software-management/dnf5/pull/656

**Comment by glum23** [comment]
oops, the test has been changed meanwhile, rebase and further changes are needed

File: dnf-behave-tests/dnf/downgrade.feature [hunk 6]
```diff
@@ -184,104 +182,51 @@ Scenario: Downgrade mixture of not available/not installed/not downgradable/down
     And stderr is
     """
     No match for argument: nosuchpkg
-    Package "wget.x86_64" of lowest version already installed, cannot downgrade it.
+    The lowest available version of the "wget.x86_64" package is already installed, cannot downgrade it
```
Comment (kontura) [inline dnf-behave-tests/dnf/downgrade.feature:185]: There is a missing dot at the end. The test doesn't pass.
 It is also missing in two other tests.

File: dnf-behave-tests/dnf/downgrade.feature [hunk 7]
```diff
@@ -25,9 +24,9 @@ Scenario: Downgrade one RPM
         | downgrade     | flac-0:1.3.3-1.fc29.x86_64                |
    When I execute dnf with args "downgrade flac"
    Then the exit code is 0
-    And stderr contains "Package flac of lowest version already installed, cannot downgrade it."
+    And stderr contains "The lowest available version of the \"flac.x86_64\" package is already installed, cannot downgrade it."
```
Comment (kontura) [inline dnf-behave-tests/dnf/downgrade.feature:27]: Can you please change to:
```gherkin
    And stderr is
    """
    The lowest available version of the "flac.x86_64" package is already installed, cannot downgrade it.
    """
```

**Comment by glum23** [comment]
@kontura hopefully done all requested changes.

**Comment by kontura** [comment]
Thank you!

**Pull request merged**